controllers/barbican_controller.go

```diff
@@ -252,52 +252,35 @@ func (r *BarbicanReconciler) reconcileNormal(ctx context.Context, instance *barb
 	Log.Info(fmt.Sprintf("TransportURL secret name %s", transportURL.Status.SecretName))
 	instance.Status.Conditions.MarkTrue(barbicanv1beta1.BarbicanRabbitMQTransportURLReadyCondition, barbicanv1beta1.BarbicanRabbitMQTransportURLReadyMessage)
 
-	//
 	// check for required OpenStack secret holding passwords for service/admin user and add hash to the vars map
-	//
-	ospSecret, hash, err := oko_secret.GetSecret(ctx, helper, instance.Spec.Secret, instance.Namespace)
+	ctrlResult, err := r.verifySecret(ctx, helper, instance, instance.Spec.Secret, []string{instance.Spec.PasswordSelectors.Service}, &configVars)
 	if err != nil {
-		if k8s_errors.IsNotFound(err) {
-			Log.Info(fmt.Sprintf("OpenStack secret %s not found", instance.Spec.Secret))
-			instance.Status.Conditions.Set(condition.FalseCondition(
-				condition.InputReadyCondition,
-				condition.RequestedReason,
-				condition.SeverityInfo,
-				condition.InputReadyWaitingMessage))
-			return ctrl.Result{RequeueAfter: time.Duration(10) * time.Second}, nil
-		}
-		instance.Status.Conditions.Set(condition.FalseCondition(
-			condition.InputReadyCondition,
-			condition.ErrorReason,
-			condition.SeverityWarning,
-			condition.InputReadyErrorMessage,
-			err.Error()))
-		return ctrl.Result{}, err
+		return ctrlResult, err
 	}
-	// Add a prefix to the var name to avoid accidental collision with other non-secret vars.
-	configVars["secret-"+ospSecret.Name] = env.SetValue(hash)
 
 	// check for Simple Crypto Backend secret holding the KEK
-	kekSecret, hash, err := oko_secret.GetSecret(ctx, helper, instance.Spec.SimpleCryptoBackendSecret, instance.Namespace)
-	if err != nil {
-		if k8s_errors.IsNotFound(err) {
-			Log.Info(fmt.Sprintf("simple Crypto backend secret %s not found", instance.Spec.SimpleCryptoBackendSecret))
-			instance.Status.Conditions.Set(condition.FalseCondition(
-				condition.InputReadyCondition,
-				condition.RequestedReason,
-				condition.SeverityInfo,
-				condition.InputReadyWaitingMessage))
-			return ctrl.Result{RequeueAfter: time.Duration(10) * time.Second}, nil
+	if len(instance.Spec.EnabledSecretStores) == 0 || slices.Contains(instance.Spec.EnabledSecretStores, barbicanv1beta1.SecretStoreSimpleCrypto) {
```

Comment:
I think with `len(instance.Spec.EnabledSecretStores) == 0` would make the `SimpleCryptoBackendSecret` a required secret, which would not be there on existing deployments and reconcile will probably stop here as the secret is missing. is that what we want? just thinking about how that'd impact the update of an existing env

Reply:
was the same with the commit which introduced the check of SimpleCryptoBackendSecret, but something we need to consider when update from fr1

Reply:
maybe should be `(len(instance.Spec.EnabledSecretStores) == 0 && SimpleCryptoBackendSecret != "") || ...`?

Reply:
ah no that won't work as it defaults to the `osp-secret`.

Reply:
SimpleCryptoBackend was the only thing available till now.  Any existing deployments would have to have had a SimpleCryptoSecret defined (or defaulted to osp-secret).  So I don't think we need to worry about existing deployments.

Reply:
could remove the default from SimpleCryptoBackendSecret to keep the current released behavior

Reply:
issue on my side not reading the code well in the stable branches :)